src/main/java/seedu/duke/module/Module.java

```diff
@@ -0,0 +1,39 @@
+package seedu.duke.module;
+
+public class Module {
+    protected String name;
+    protected String id;
+    protected String semester;
+    protected boolean isName;
```

Comment:
Hello, could you explain the variables `isName` and `isId`? I don't understand why it is a `boolean` type instead of a `String` type. Thank you.

Reply:
Because users sometimes may add some modules which are not in the database and they may just input one module's name or id. So I use these two `boolean` variables `isName` and `isId` to indicate whether one module's name and id is valid. Maybe the two variables' name is confusing. I will change them into `isNameValid` and `isIdValid`.